File: src/types/audit.ts
import { UserPersona } from './persona';

export interface AuditLogItem {
  id: string;
  timestamp: string;
  user_persona: UserPersona;
  action: string;
  data_source: string;
  status: 'SUCCESS' | 'WARNING' | 'FAILED' | string;
  user_prompt: string;
  execution_time_ms: number;
  summary?: string;
}

File: backend/app/models/audit.py
from pydantic import BaseModel
from typing import Optional

class AuditModel(BaseModel):
    audit_id: str
    user_persona: str
    action: str
    data_source: str
    status: str
    execution_time_ms: int
    user_prompt: str
    summary: Optional[str] = None
    created_at: str
